### PR TITLE
Aria required and aria invalid input attributes

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #12399

Replaces #12647

### CODE REVIEW

File: framework/assets/yii.activeForm.js [hunk 1]
```diff
@@ -775,4 +778,9 @@
         }
     };
 
+    var toggleAriaInvalid = function($form, attribute, hasError) {
```
Comment (SilverFire) [inline framework/assets/yii.activeForm.js]: `function ($form`

File: framework/CHANGELOG.md [hunk 2]
```diff
@@ -60,7 +61,7 @@ Yii Framework 2 Change Log
 2.0.10 October 20, 2016
 -----------------------
 
-- Bug #7670: Added `yii\web\UrlNormalizer` for normalizing requests with and without trailing slashes (rob006, cronfy, klimov-paul)
+- Bug #7670: Added `yii\web\UrlNormalizer` for normalizing requests with and without trailing slashes (rob006, cronfy, klimov-paul)
```
Comment (SilverFire) [inline framework/CHANGELOG.md]: Extra space was added

**Review by SilverFire — APPROVED** [review]
Good except minor comments bellow

File: framework/widgets/ActiveField.php [hunk 3]
```diff
@@ -789,10 +815,32 @@ protected function getClientOptions()
...
 
     /**
+     * Checks if client validation enabled for the field
+     * @return bool
+     * @since 2.0.11
+     */
+    protected function getEnableClientValidation()
+    {
+        return $this->enableClientValidation
+            || $this->enableClientValidation === null && $this->form->enableClientValidation;
```
Comment (dynasource) [inline framework/widgets/ActiveField.php]: this line break is not really consistent with the next method

File: framework/widgets/ActiveField.php [hunk 4]
```diff
@@ -158,6 +163,11 @@ class ActiveField extends Component
      */
     private $_skipLabelFor = false;
 
+    /**
+     * @var bool whether `aria-invalid` HTML attribute should be toggled by validation on client
+     */
+    private $_skipClientAriaInvalid = true;
```
Comment (dynasource) [inline framework/widgets/ActiveField.php]: this naming is a bit unclear

**Review by dynasource — CHANGES_REQUESTED** [review]
Looks good overall. Minor remark. 
1. The property _skipClientAriaInvalid should get a better name. Its unclear  what its objective is
2. Shouldnt this PR be combined with adjustLabelFor in to 1 options processing method (perhabs even including the array_merge part

File: framework/widgets/ActiveField.php (hunk 4 above)
Comment (samdark) [inline framework/widgets/ActiveField.php]: AriaInvalid is an attribute name in HTML.

Comment (dynasource) [inline framework/widgets/ActiveField.php]: it doesnt really match it's description

**Review by dynasource — APPROVED** [review]

File: framework/widgets/ActiveField.php [hunk 5]
```diff
@@ -789,10 +815,31 @@ protected function getClientOptions()
...
+            'updateAriaInvalid' => true,
         ]);
     }
 
     /**
+     * Checks if client validation enabled for the field
+     * @return bool
+     * @since 2.0.11
+     */
+    protected function clientValidationEnabled()
```
Comment (cebe) [inline framework/widgets/ActiveField.php]: as this returns bool, a better name would be `isClientValidationEnabled()`.

File: framework/widgets/ActiveField.php [hunk 6]
```diff
@@ -789,10 +815,31 @@ protected function getClientOptions()
...
+    {
+        return $this->enableClientValidation || $this->enableClientValidation === null && $this->form->enableClientValidation;
+    }
+
+    /**
+     * Checks if ajax validation enabled for the field
+     * @return bool
+     * @since 2.0.11
+     */
+    protected function ajaxValidationEnabled()
```
Comment (cebe) [inline framework/widgets/ActiveField.php]: as this returns bool, a better name would be `isAjaxValidationEnabled()`.

File: framework/widgets/ActiveField.php [hunk 7]
```diff
@@ -158,6 +163,11 @@ class ActiveField extends Component
      */
     private $_skipLabelFor = false;
 
+    /**
+     * @var bool whether `aria-invalid` HTML attribute should be updated by client validation
```
Comment (cebe) [inline framework/widgets/ActiveField.php]: why should that be necessary? If client validation finds that the error is gone it should update it, no?

**Review by cebe — CHANGES_REQUESTED** [review]

**Review by cebe — APPROVED** [review]

**Pull request merged**